parsers/US_CA.py

```diff
@@ -34,37 +34,51 @@ def fetch_production(country_code='US-CA', session=None):
       'source': 'mysource.com'
     }
     """
-    # caiso.com provides daily data until the day before today
-    # get a clean date at the beginning of yesterday
-    yesterday = arrow.utcnow().to('US/Pacific').shift(days=-1).replace(hour = 0, minute = 0, second = 0, microsecond = 0)
-    url = 'http://content.caiso.com/green/renewrpt/' + yesterday.format('YYYYMMDD') +'_DailyRenewablesWatch.txt'
-
-    renewableResources = pandas.read_table(url, sep='\t\t', skiprows=2, header=None, names=['Hour', 'GEOTHERMAL','BIOMASS', 'BIOGAS', 'SMALL HYDRO','WIND TOTAL', 'SOLAR PV', 'SOLAR THERMAL'], skipfooter=27, skipinitialspace=True, engine='python')
-    otherResources = pandas.read_table(url, sep='\t\t', skiprows=30, header=None, names=['Hour', 'RENEWABLES', 'NUCLEAR', 'THERMAL', 'IMPORTS', 'HYDRO'], skipinitialspace=True, engine='python')
-
-    dailyData = []
+    url = 'http://www.caiso.com/outlook/SP/fuelsource.csv'
+    csv = pandas.read_csv(url)
+    latest_index = len(csv)-1
+    h, m = map(int, csv['Time'][latest_index].split(':'))
+    date = arrow.utcnow().to('US/Pacific').replace(hour = h, minute = m, second = 0, microsecond = 0)
```

Comment:
`utcnow` and the time of the csv might not be the same date.
Can't you directly parse `csv['Time']`?

Reply:
No. csv['Time'] only countains the time of the day (hour and minute). That's why I'm using the date "now" in US/Pacific time, which should be the same as the latest value from this file. Except if this is called somewhere between 00:00 and 00:05 in California, then the latest value might be from the day before and the date will indeed be wrong...

Reply:
Actually the date is not in the fuelsource csv file but it is in other files like CAISO-renewables. I can use the date from that file instead.